File: src/main/java/common/block/Block.java
package common.block;

import common.world.World;

public abstract class Block {
    protected String name;
    private float hardness, blastResistance = 1.0f;
    private boolean solid = true;

    public Block(String name){
        this.name = name;
    }

    /** These two functions will CRASH the server because they will be calling client code server side, when that
     * Client code won't exist server side!
     *
     * Additionally, they need to pass the actual block with the actual positions.
     * **/
    @Deprecated
    public abstract void onBlockRightClick();

    @Deprecated
    public abstract void onBlockCollision();

    public abstract void onTick(World world, int x, int y);

    public abstract String getName();
}

File: src/main/java/common/block/Block.kt
package common.block

import common.registries.RegistryNameable
import common.world.World

abstract class Block(name: String): RegistryNameable(name) {
    var name: String
        protected set
    private val hardness: Float = 0.toFloat()
    private val blastResistance = 1.0f
    private val solid = true

    init {
        this.name = name
    }

    /** These two functions will CRASH the server because they will be calling client code server side, when that
     * Client code won't exist server side!
     *
     * Additionally, they need to pass the actual block with the actual positions.
     */
    @Deprecated("")
    abstract fun onBlockRightClick()

    @Deprecated("")
    abstract fun onBlockCollision()

    abstract fun onTick(world: World, x: Int, y: Int)
}
